itou/immersion_facile/utils.py

```diff
@@ -0,0 +1,28 @@
+from urllib.parse import quote, urlencode
+
+from itou.utils.constants import IMMERSION_FACILE_SITE_URL
+
+
+def immersion_facile_search_url(user):
+    """
+    :return: a search URL on Immersion Facilitée's site for parameterized user
+    """
+    params = {}
+
+    # in testing, there are some differences between how addresses are validated between
+    # IF's service and ours, so only we can only consider the geolocation reliable if
+    # lat/lng is present
+    if user.coords:
+        params["distanceKm"] = 20
+        params["latitude"] = user.latitude
+        params["longitude"] = user.longitude
+        params["sortedBy"] = "distance"
+
+        address_parts = [user.city, user.region, "France"]
```

Comment:
C'est un chantier en cours donc les deux existent en parallèle mais le but est de n'avoir plus que `insee_city` car la donnée est plus correcte (et cohérente) que les champs ouvert `user.city` et `user.department` qui peuvent contenir un peu tout et n'importe quoi :
```suggestion
        if user.insee_city:
            address_parts = [user.insee_city.name, user.insee_city.region, "France"]
        else:
            address_parts = [user.city, user.region, "France"]
```
C'est peut-être l'explication pour les différences que tu mentionnes dans ton commentaire au dessus, je ne sais pas si `place` peux être envoyé sans les 4 autres mais je dirais que si on utilise `insee_city` alors `place` pourrais être présent même sans `user.coords`, même si ça devrais jamais arriver (:crossed_fingers:).

Reply:
Si ça devrait jamais arriver je préfères laisser la condition sur `user.coords` - c'est plus simple je pense. Sans geolocation leur page va montrer la ville et la distance précisées sans appliquer les filtres donc ça pourrait poser un problème UX